packages/core-bridge/ts/worker-tuner.ts

```diff
@@ -69,3 +87,102 @@ export interface FixedSizeSlotSupplier {
   // The maximum number of slots that can be issued
   numSlots: number;
 }
+
+/**
+ * The interface can be implemented to provide custom slot supplier behavior.
+ */
+export interface CustomSlotSupplier<SI extends SlotInfo> {
+  type: 'custom';
+
+  /**
+   * This function is called before polling for new tasks. Your implementation should block until a
+   * slot is available then return a permit to use that slot.
+   *
+   * The only acceptable exception to throw is AbortError, any other exceptions thrown will be
+   * logged and ignored.
+   *
+   * @param ctx The context for slot reservation.
+   * @param abortSignal The SDK may decide to abort the reservation request if it's no longer
+   *   needed. Implementations may clean up and then must reject the promise with AbortError.
+   * @returns A permit to use the slot which may be populated with your own data.
+   */
+  reserveSlot(ctx: SlotReserveContext, abortSignal: AbortSignal): Promise<SlotPermit>;
+
+  /**
+   * This function is called when trying to reserve slots for "eager" workflow and activity tasks.
```

Comment:
There's no case of eager workflow task in TS at this point.

Reply:
I think I'll leave it as is for when we maybe reintroduce